enhancements/installer/infrastructure-apis-custom-trust.md

```diff
@@ -0,0 +1,236 @@
+
+
+## Motivation
+
+OpenShift components interact quite a bit with the underlying infrastructure using APIs to provide it's users various features and a majority of our operators today interact with infrastructure APIs. For our commercial platforms like AWS, Azure, GCP's public cloud trusting the secured API endpoints is not a problem, but for other platforms like vSphere, OpenStack or even private clouds from AWS or Azure we need our operators to trust user-provided bundles to securely communicate to the APIs.
+
+### Goals
+
+1. Allow users to provide custom trust for APIs using openshift-installer.
+2. Allow users to provide custom trust for APIs as a day-2 operation.
+3. Provide a no hassle workflow for operators to consume the trust bundle on-par with [proxy][egress-proxy-proposal]
+
+### Non-Goals
+
+None at this time.
+
+## Proposal
+
+1. openshift-installer allows the users to provide custom trust bundle for infrastructure APIs using the `install-config.yaml`.
+2. [Infrastructure][openshift-api-infrastructure] object allows user to provide custom trust in the form of a ConfigMap in `openshift-config` namespace.
+3. A controller validates the user-provided ConfigMap and provides it as an API for consumption in the form of a ConfigMap in `openshift-config-managed` namespace.
+4. The [proxy controller][cno-proxy-controller] uses the above ConfigMap another source for generating the `openshift-config-managed/trusted-ca-bundle` so that operators get the infrastructure APIs trust when using the [`config.openshift.io/inject-trusted-cabundle`][egress-proxy-proposal] mechanism.
+
+### User Stories
+
+#### AWS GovCloud testing using emulation
+
+#### AWS C2S/SC2S
+
+#### Azure Stack Hub
+
+### Risks and Mitigations
+
+None
+
+## Design Details
+
+### Can we configure the custom trust for SDKs easily?
+
+There are 2 types of ways we would need to configure custom trust for infrastructure APIs,
+
+- The operator itself is talking to the APIs so it needs to configure the SDK.
+- The operator is managing a component that communicates with infrastructure APIs so it needs to configure that component to use the custom trust.
+
+#### SDKs directly
+
+This is not trivial, as some Go SDKs require the user to configure the http.Client with appropiate transport like for example AWS and GCP, [for example](https://github.com/aws/aws-sdk-go/issues/2404#issuecomment-454196045). While there are Go SDKs like for Azure that do not provide any great way, [for example](https://github.com/Azure/azure-sdk-for-go/issues/5974).
+
+Also since configuring the trust at best requires creating custom http transport, this is difficult and error prone to the same extent that made us move to using PROXY environment variable instead of managing transports for http clients with these specific values. Using universal system trust mechanism is more safer for the same reasons.
```

Comment:
Using universal system trust mechanism is more safer for the same reasons. -- this claim requires evidence.  I would actually say that conflating trust in this way produces unexpected outcomes where less trustworthy CAs can accidentally be used to verify servers they were not intended to verify.

Taking the proxy as a for instance, as a user, I would be very surprised if the CA bundle I thought would be used to secure communication to my cloud provider was suddenly valid to terminate connections to my company's IdP or connections to github.com.

This claim doesn't appear to be correct to me.